spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/integration/StompMessageBuilderHelper.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.integration;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.Trace;
+import org.springframework.cloud.sleuth.trace.TraceContextHolder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.simp.SimpMessageHeaderAccessor;
+import org.springframework.messaging.simp.SimpMessageType;
+
+/**
+ * Helper class to create STOMP message
+ * @author Gaurav Rai Mazra
+ *
+ */
+public class StompMessageBuilderHelper {
+	public static StompMessageBuilder fromMessage(Message<?> message) {
+		return new StompMessageBuilder(message);
+	}
+	
+	public static void addAnnotationsToSpanFromMessage(Message<?> message, Span span) {
+		for (Map.Entry<String, Object> entry : message.getHeaders().entrySet()) {
+			if (!Trace.HEADERS.contains(entry.getKey())) {
+				String key = "/messaging/headers/" + entry.getKey().toLowerCase();
+				String value = entry.getValue() == null ? null : entry.getValue().toString();
+				span.addAnnotation(key, value);
+			}
+		}
+		
+		Object payload = message.getPayload();
+		if (payload != null) {
+			span.addAnnotation("/messaging/payload/type", payload.getClass().getCanonicalName());
+			
+			if (payload instanceof String) {
+				span.addAnnotation("/messaging/payload/size", String.valueOf(((String)payload).length()));
```

Comment:
Before that there is payload type is also shown. So, it might not create any confusion. I am not sure if encoding type in case of String need to be shown as annotation.